wasm/src/programs/authorized.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the Aleo SDK library.
+
+// The Aleo SDK library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Aleo SDK library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Aleo SDK library. If not, see <https://www.gnu.org/licenses/>.
+
+/// The API URL for the network.
+pub const API_URL: &str = "https://api.explorer.aleo.org/v1/testnet3/explorer";
+
+use crate::types::{AuthorizationNative, TransactionNative};
+use snarkvm_wasm::utilities::serialize::DeserializeExt;
+
+use serde::{de, ser::SerializeStruct, Deserialize, Deserializer, Serialize, Serializer};
+
+pub(crate) struct Authorized {
+    /// The authorization for the main function execution.
+    function: AuthorizationNative,
+    /// The authorization for the fee execution.
+    fee: Option<AuthorizationNative>,
+    /// Whether to broadcast the transaction.
+    broadcast: bool,
+}
+
+impl Authorized {
+    /// Initializes a new authorization.
+    pub const fn new(function: AuthorizationNative, fee: Option<AuthorizationNative>, broadcast: bool) -> Self {
+        Self { function, fee, broadcast }
+    }
+
+    /// Executes the authorization, returning the resulting transaction.
+    pub async fn execute(authorization: &Self) -> Result<TransactionNative, String> {
+        // Execute the authorization.
+        let response = reqwest::Client::new()
+            .post(format!("{API_URL}/execute"))
+            .header("Content-Type", "application/json")
+            .body(serde_json::to_string(authorization).map_err(|e| e.to_string())?)
+            .send()
+            .await
+            .map_err(|e| e.to_string())?;
+
+        // Ensure the response is successful.
+        match response.status().is_success() {
+            // Return the transaction.
+            true => Ok(response.json().await.map_err(|e| e.to_string())?),
+            // Return the error.
+            false => Err(response.text().await.map_err(|e| e.to_string())?),
+        }
+    }
+}
+
+impl Serialize for Authorized {
```

Comment:
Instead of manually implementing Serialize/Deserialize, it would be better to use attributes to control the serialization:

```rust
#[derive(Serialize, Deserialize)]
pub(crate) struct Authorized {
    /// The authorization for the main function execution.
    function: AuthorizationNative,

    /// The authorization for the fee execution.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    fee: Option<AuthorizationNative>,

    /// Whether to broadcast the transaction.
    broadcast: bool,
}
```

Reply:
Yep you're %100 right